syntax = "proto3";
package NetworkData;
import "google/protobuf/timestamp.proto";

message NetworkData {
  string ip = 1;
  string uuid = 2;
  ENetworkType type = 3;
  uint64 bufSize = 4;
  string data = 5;
  google.protobuf.Timestamp timestamp = 6;
}

enum ENetworkType {
  LOGIN = 0;
  REGISTER = 1;
  RETRIEVE = 2;
  ACCESS = 3;
  REJECT = 4;
  LOGOUT = 5;
  COUNT_FOR_USER_DATA = 6;

  MOVE = 100;
  ATTACK = 101;
  DROP_ITEM = 102;
  USE_ITEM = 103;
  USE_SKILL = 104;
  COUNT_FOR_LOGIC_DATA = 105;

  SYSTEM_OPTION_1 = 200;
  SYSTEM_OPTION_2 = 201;
  SYSTEM_OPTION_3 = 202;
  COUNT_FOR_SYSTEM_DATA = 203;

  ADMIN_SERVER_OFF = 900;
  ADMIN_SERVER_ON = 901;
  ADMIN_SERVER_REBOOT = 902;
  COUNT_FOR_ADMIN_DATA = 903;
}

enum ELoginError {
    USER_NOT_FOUND = 0;
    USER_ALREADY_EXIST = 1;
    INCORRECT = 2;
}